src/index.js

```diff
@@ -1,57 +1,56 @@
-const fetch = require("node-fetch")
 const {getColor, getRandomColor} = require("./colors")
-
 class Hue {
-    constructor(ip, username) {
-        this.api = `http://${ip}/api/${username}`
+    constructor(hueClient) {
+        this.client = hueClient
     }
 
-    readLight = async id => {
-        const response = await fetch(`${this.api}/lights/${id}`, {
-            method: "GET",
-        })
-
-        const json = await response.json()
-        const light = {id, ...json}
-
-        return light
+    getLightInfo = async id => {
+        return this.client.readLight(id)
     }
 
-    readLights = async () => {
-        const response = await fetch(`${this.api}/lights`, {method: "GET"})
-        const json = await response.json()
-
-        const lights = Object.entries(json).map(([id, light]) => {
-            return {id, ...light}
-        })
+    getLightsInfo = async () => {
+        return this.client.readLights(this.client)
+    }
 
-        return lights
+    turnOnLight = id => {
+        this.client.updateLight(id, {on: true}, )
     }
 
-    updateLight = (id, state) => {
-        fetch(`${this.api}/lights/${id}/state`, {
-            method: "PUT",
-            body: JSON.stringify(state),
-        })
+    turnOffLight = id => {
+        this.client.updateLight(id, {on: false}, )
+    }
+    
+    blinkLight = async id => {
+        this.turnOnLight(id)
+        await new Promise(resolve => setTimeout(resolve, 750))
+        this.turnOffLight(id)
+        await new Promise(resolve => setTimeout(resolve, 750))
+        this.turnOnLight(id)
     }
 
-    turnOnLight = id => {
-        this.updateLight(id, {on: true})
+    turnOffAllLights = async () => {
+        const lights = await this.getLightsInfo()
+        lights.forEach(async (light) => {
+            await this.client.updateLight(light.id, {on: false})
```

Comment:
Let's remove `async` for now and just fire off the requests.

Also, I would prefer to use `turnOffLight(light.id)` instead of `updateLight`.